---
title: "High-Dimensional Mediation Analysis"
subtitle: "A Guide to Using the HIMA Package"
author: "The HIMA Development Team"
date: "`r Sys.Date()`"
output:
  rmarkdown::html_vignette:
    toc: true
    toc_depth: 3
vignette: >
  %\VignetteIndexEntry{High-Dimensional Mediation Analysis}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  eval = TRUE,
  echo = TRUE # Ensures all code is displayed by default
)
library(HIMA)
```

# Introduction

Mediation analysis is a statistical methodology employed to investigate the underlying mechanisms through which an independent variable influences a dependent variable through an intermediary variable (mediator). This approach typically involves evaluating indirect, direct, and total effects within a specified statistical framework. A key objective of mediation analysis is to test the mediation hypothesis, which suggests that the impact of the independent variable on the dependent variable is transmitted through one or more mediating variables. By investigating these pathways, we can achieve a more comprehensive understanding of the causal relationships among the variables. For instance, mediation analysis can enhance the refinement of interventions to improve their efficacy in clinical trials. In observational studies, mediation analysis can serve to pinpoint intervention targets and elucidate the underlying mechanisms of diseases.


# Package Overview

The `HIMA` package provides tools for estimating and testing high-dimensional mediation effects, tailored for modern `omic` data such as epigenetics, transcriptomics, and microbiomics. The function, `hima`, serves as a comprehensive wrapper function designed to implement various high-dimensional mediation analysis methods for both estimating and testing mediation effects. `hima` can automatically select the appropriate method based on the outcome and mediator data type. The `hima` function is designed to be extensible, allowing future features and methods to seamlessly integrate within its framework, ensuring a consistent and user-friendly experience. 

### Key Features

- High-dimensional mediation analysis for continuous, binary, and survival outcomes.
- Specialized methods for compositional microbiome and quantile mediation analysis.
- Flexible penalty options including `DBlasso`, `MCP`, and `SCAD`.
- High-level wrapper function `hima` for easy implementation.


# Package Installation

HIMA is now available in the R CRAN repository. To install from CRAN:

```r
install.packages("HIMA")
```

To install the development version from GitHub:

```r
library(devtools)
install_github("yinanzheng/HIMA")
```

If the package "qvalue" is not found, please first install the "qvalue" package through Bioconductor:

```r
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("qvalue")
```


# Data Preparation and Settings

To use the `hima` function, ensure your data is prepared according to the following guidelines:

### 1. Formula Argument (`formula`)

Define the model formula to specify the relationship between the outcome, treatment, and covariates. Ensure the following:
- **General Form**: Use the format `Outcome ~ Exposure + Covariates`. Note that the `Exposure` variable represents the exposure of interest (e.g., "Treatment" in the demo examples) and it has to be listed as the first independent variable in the formula. `Covariates` are optional.
- **Interaction Terms**: Optionally include interaction terms if required by your analysis.
- **Survival Data**: For survival analysis, use the format `Surv(time, event) ~ Exposure + Covariates`. See data examples `SurvivalData$PhenoData` for more details.

### 2. Phenotype Data (`data.pheno`)

The `data.pheno` object should be a `data.frame` or `matrix` containing the phenotype information for the analysis. Key requirements include:
- **Rows:** Represent samples.
- **Columns:** Include variables such as the outcome, treatment, and optional covariates.
- **Formula Consistency:** Ensure that all variables specified in the `formula` argument (e.g., `Outcome`, `Treatment`, and `Covariates`) are present in `data.pheno`.

### 3. Mediator Data (`data.M`)

The `data.M` object should be a `data.frame` or `matrix` containing high-dimensional mediators. Key requirements include:
- **Rows:** Represent samples, aligned with the rows in `data.pheno`.
- **Columns:** Represent mediators (e.g., CpGs, genes, or other molecular features).
- **Mediator Type:** Specify the type of mediators in the `mediator.type` argument. Supported types include (more supporting types will be added in the future):
  - `"gaussian"` for continuous mediators (default, e.g., DNA methylation data).
  - `"negbin"` for count data (e.g., transcriptomic data).
  - `"compositional"` for microbiome or other compositional data.

### 4. About data scaling

In most real-world data analysis scenarios, `scale` is typically set to `TRUE`, ensuring that the exposure (variable of interest), mediators, and covariates (if included) are standardized to a mean of zero and a variance of one. However, if your data is already pre-standardized—such as in simulation studies or when using our demo dataset-`scale` should be set to `FALSE` to prevent introducing biases or altering the original data structure. When applying `HIMA` to simulated data, if `scale` is set to `TRUE`, it is imperative to preprocess the mediators by scaling them to have a mean of zero and a variance of one prior to generating the outcome variables.



# Examples

## Load the `HIMA` Package

```{r load-HIMA}
library(HIMA)
```

## Continuous Outcome Analysis

We begin with an example of analyzing continuous outcomes using the `HIMA` package. The following code snippet demonstrates how to perform a high-dimensional mediation analysis with continuous outcomes:

```{r continuous-example}
data(ContinuousOutcome)
hima_continuous.fit <- hima(
  Outcome ~ Treatment + Sex + Age,
  data.pheno = ContinuousOutcome$PhenoData,
  data.M = ContinuousOutcome$Mediator,
  mediator.type = "gaussian",
  penalty = "MCP",
  scale = TRUE
)
summary(hima_continuous.fit)
```

We can use the `summary` function with the `desc = TRUE` option to show the description of the output results:

```{r summary-desc}
summary(hima_continuous.fit, desc=TRUE)
```

## Efficient HIMA

For continuous mediators and outcomes, use the efficient HIMA method with the `efficient = TRUE` option. This method is computationally efficient for large datasets and high-dimensional mediators. For example, consider the following code snippet for an efficient HIMA analysis:

```{r efficient-example}
hima_efficient.fit <- hima(
  Outcome ~ Treatment + Sex + Age,
  data.pheno = ContinuousOutcome$PhenoData,
  data.M = ContinuousOutcome$Mediator,
  mediator.type = "gaussian",
  efficient = TRUE,
  penalty = "MCP",
  scale = TRUE
)
summary(hima_efficient.fit)
```

## Binary Outcome Analysis

The package can also handle binary outcomes. Here is an example of binary outcome analysis using `HIMA`:   

```{r binary-example}
data(BinaryOutcome)
hima_binary.fit <- hima(
  Disease ~ Treatment + Sex + Age,
  data.pheno = BinaryOutcome$PhenoData,
  data.M = BinaryOutcome$Mediator,
  mediator.type = "gaussian",
  penalty = "MCP",
  scale = TRUE
)
summary(hima_binary.fit)
```

## Survival Outcome Analysis

For survival data, `HIMA` incorporates a Cox proportional hazards approach. Here is an example of survival outcome analysis using `HIMA`:

```{r survival-example}
data(SurvivalData)
hima_survival.fit <- hima(
  Surv(Time, Status) ~ Treatment + Sex + Age,
  data.pheno = SurvivalData$PhenoData,
  data.M = SurvivalData$Mediator,
  mediator.type = "gaussian",
  penalty = "DBlasso",
  scale = TRUE
)
summary(hima_survival.fit)
```

## Microbiome Mediation Analysis

For compositional microbiome data, `HIMA` employs specialized transformations. Here is an example of microbiome mediation analysis using `HIMA`:

```{r microbiome-example}
data(MicrobiomeData)
hima_microbiome.fit <- hima(
  Outcome ~ Treatment + Sex + Age,
  data.pheno = MicrobiomeData$PhenoData,
  data.M = MicrobiomeData$Mediator,
  mediator.type = "compositional",
  penalty = "DBlasso"
)
summary(hima_microbiome.fit)
```

## Quantile Mediation Analysis

Perform quantile mediation analysis using the `quantile = TRUE` option. Here is an example of quantile mediation analysis using `HIMA`:

```{r quantile-example}
data(QuantileData)
hima_quantile.fit <- hima(
  Outcome ~ Treatment + Sex + Age,
  data.pheno = QuantileData$PhenoData,
  data.M = QuantileData$Mediator,
  mediator.type = "gaussian",
  quantile = TRUE,
  penalty = "MCP",
  tau = c(0.3, 0.5, 0.7),
  scale = TRUE
)
summary(hima_quantile.fit)
```

# References

1. Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L,
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies.
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171; PMCID: PMC5048064

2. Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators.
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID: PMC8570823

3. Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation Effect Selection in High-dimensional and Compositional Microbiome data.
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955

4. Zhang H, Chen J, Li Z, Liu L. Testing for Mediation Effect with Application to Human Microbiome Data.
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450

5. Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. HIMA2: High-dimensional Mediation Analysis and Its Application in
Epigenome-wide DNA Methylation Data. BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002

6. Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI: 10.1093/bioinformatics/btae055. PMID: 38290773; PMCID: PMC10873903

7. Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI: 10.1007/s12561-024-09447-4.

# Session Information

```{r session-info}
sessionInfo()
```
