<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>High-dimensional Mediation Analysis — hima • HIMA</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="High-dimensional Mediation Analysis — hima"><meta name="description" content="hima is a wrapper function designed to perform various HIMA methods for estimating and testing high-dimensional mediation effects.
hima can automatically select the appropriate HIMA method based on the outcome and mediator data type."><meta property="og:description" content="hima is a wrapper function designed to perform various HIMA methods for estimating and testing high-dimensional mediation effects.
hima can automatically select the appropriate HIMA method based on the outcome and mediator data type."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HIMA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.3.3</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/hima-vignette.html">Tutorials</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/YinanZheng/HIMA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>High-dimensional Mediation Analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/YinanZheng/HIMA/blob/HEAD/R/hima.R"><code>R/hima.R</code></a></small>
      <div class="d-none name"><code>hima.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>hima</code> is a wrapper function designed to perform various HIMA methods for estimating and testing high-dimensional mediation effects.
<code>hima</code> can automatically select the appropriate HIMA method based on the outcome and mediator data type.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">hima</span><span class="op">(</span></span>
<span>  <span class="va">formula</span>,</span>
<span>  <span class="va">data.pheno</span>,</span>
<span>  <span class="va">data.M</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gaussian"</span>, <span class="st">"negbin"</span>, <span class="st">"compositional"</span><span class="op">)</span>,</span>
<span>  penalty <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"DBlasso"</span>, <span class="st">"MCP"</span>, <span class="st">"SCAD"</span>, <span class="st">"lasso"</span><span class="op">)</span>,</span>
<span>  quantile <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  efficient <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  longitudinal <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  id.var <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sigcut <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  contrast <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  subset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncore <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-formula">formula<a class="anchor" aria-label="anchor" href="#arg-formula"></a></dt>
<dd><p>an object of class <code>formula</code> representing the overall effect model to be fitted, specified as <code>outcome ~ exposure + covariates</code>.
The "exposure" variable (the variable of interest) must be listed first on the right-hand side of the formula.
For survival outcomes specified using <code>Surv()</code>, the exposure should be the first
variable after the <code><a href="https://rdrr.io/r/base/tilde.html" class="external-link">~</a></code>.</p></dd>


<dt id="arg-data-pheno">data.pheno<a class="anchor" aria-label="anchor" href="#arg-data-pheno"></a></dt>
<dd><p>a data frame containing the exposure, outcome, and covariates specified in the formula. Variable names in <code>data.pheno</code> must match those
in the formula. When <code>scale = TRUE</code>, the exposure and covariates will be scaled (the outcome retains its original scale).</p></dd>


<dt id="arg-data-m">data.M<a class="anchor" aria-label="anchor" href="#arg-data-m"></a></dt>
<dd><p>a <code>data.frame</code> or <code>matrix</code> of high-dimensional mediators, with rows representing samples and columns representing mediator variables.
When <code>scale = TRUE</code>, <code>data.M</code> will be scaled.</p></dd>


<dt id="arg-mediator-type">mediator.type<a class="anchor" aria-label="anchor" href="#arg-mediator-type"></a></dt>
<dd><p>a character string indicating the data type of the high-dimensional mediators (<code>data.M</code>). Options are: <code>'gaussian'</code> (default):
for continuous mediators. <code>'negbin'</code>: for count data mediators modeled using the negative binomial distribution (e.g., RNA-seq data). <code>'compositional'</code>:
for compositional data mediators (e.g., microbiome data).</p></dd>


<dt id="arg-penalty">penalty<a class="anchor" aria-label="anchor" href="#arg-penalty"></a></dt>
<dd><p>a character string specifying the penalty method to apply in the model. Options are: <code>'DBlasso'</code>: De-biased LASSO (default). <code>'MCP'</code>:
Minimax Concave Penalty. <code>'SCAD'</code>: Smoothly Clipped Absolute Deviation. <code>'lasso'</code>: Least Absolute Shrinkage and Selection Operator. Note: Survival HIMA and microbiome
HIMA can only be performed with <code>'DBlasso'</code>. Quantile HIMA and efficient HIMA cannot use <code>'DBlasso'</code>; they always apply <code>'MCP'</code>.</p></dd>


<dt id="arg-quantile">quantile<a class="anchor" aria-label="anchor" href="#arg-quantile"></a></dt>
<dd><p>logical. Indicates whether to use quantile HIMA (<code>hima_quantile</code>). Default is <code>FALSE</code>. Applicable only for classic HIMA with a continuous outcome and
<code>mediator.type = 'gaussian'</code>. If <code>TRUE</code>, specify the desired quantile(s) using the <code>tau</code> parameter; otherwise, the default <code>tau = 0.5</code> (i.e., median) is used.</p></dd>


<dt id="arg-efficient">efficient<a class="anchor" aria-label="anchor" href="#arg-efficient"></a></dt>
<dd><p>logical. Indicates whether to use efficient HIMA (<code>hima_efficient</code>). Default is <code>FALSE</code>. Applicable only for classic HIMA with a continuous outcome and
<code>mediator.type = 'gaussian'</code>.</p></dd>


<dt id="arg-longitudinal">longitudinal<a class="anchor" aria-label="anchor" href="#arg-longitudinal"></a></dt>
<dd><p>logical. Indicates whether to run the longitudinal survival mediation model <code>hima_survival_long</code> (requires a <code>Surv(tstart, tstop, status)</code> outcome).
Default = <code>FALSE</code>.</p></dd>


<dt id="arg-id-var">id.var<a class="anchor" aria-label="anchor" href="#arg-id-var"></a></dt>
<dd><p>Character string specifying the column name in <code>data.pheno</code> that stores subject identifiers. Required when <code>longitudinal = TRUE</code>.</p></dd>


<dt id="arg-scale">scale<a class="anchor" aria-label="anchor" href="#arg-scale"></a></dt>
<dd><p>logical. Determines whether the function scales the data (exposure, mediators, and covariates). Default is <code>TRUE</code>. Note: For simulation studies, set
<code>scale = FALSE</code> to avoid estimate compression (i.e., shrinkage of estimates toward zero due to scaling).</p></dd>


<dt id="arg-sigcut">sigcut<a class="anchor" aria-label="anchor" href="#arg-sigcut"></a></dt>
<dd><p>numeric. The significance cutoff for selecting mediators. Default is <code>0.05</code>.</p></dd>


<dt id="arg-contrast">contrast<a class="anchor" aria-label="anchor" href="#arg-contrast"></a></dt>
<dd><p>a named list of contrasts to be applied to factor variables in the covariates (cannot be the variable of interest).</p></dd>


<dt id="arg-subset">subset<a class="anchor" aria-label="anchor" href="#arg-subset"></a></dt>
<dd><p>an optional vector specifying a subset of observations to use in the analysis.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>logical. Determines whether the function displays progress messages. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-parallel">parallel<a class="anchor" aria-label="anchor" href="#arg-parallel"></a></dt>
<dd><p>logical. Enable parallel computing feature? Default = <code>FALSE</code>.</p></dd>


<dt id="arg-ncore">ncore<a class="anchor" aria-label="anchor" href="#arg-ncore"></a></dt>
<dd><p>number of cores to run parallel computing Valid when <code>parallel = TRUE</code>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>reserved passing parameter (or for future use).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data.frame containing mediation testing results of selected mediators.</p><dl><dt>ID: </dt>
<dd><p>Mediator ID/name.</p></dd>

    <dt>alpha: </dt>
<dd><p>Coefficient estimates of exposure (X) –&gt; mediators (M) (adjusted for covariates).</p></dd>

    <dt>beta: </dt>
<dd><p>Coefficient estimates of mediators (M) –&gt; outcome (Y) (adjusted for covariates and exposure).</p></dd>

    <dt>alpha*beta: </dt>
<dd><p>The estimated indirect (mediation) effect of exposure on outcome through each mediator.</p></dd>

    <dt>rimp: </dt>
<dd><p>Relative importance- the proportion of each mediator's mediation effect relative to the sum of the absolute mediation effects of all significant mediators.</p></dd>

    <dt>p-value: </dt>
<dd><p>The joint p-value assessing the significance of each mediator's indirect effect, calculated based on the corresponding statistical approach.</p></dd>

    <dt>tau: </dt>
<dd><p>The quantile level of the outcome (applicable only when using the quantile mediation model).</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>1. Zhang H, Zheng Y, Hou L, Liu L, HIMA: An R Package for High-Dimensional Mediation Analysis. Journal of Data Science. 2025. DOI: 10.6339/25-JDS1192</p>
<p>2. Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W, Schwartz J, Just A, Colicino E, Vokonas P, Zhao L,
Lv J, Baccarelli A, Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects in Epigenetic Studies.
Bioinformatics. 2016. DOI: 10.1093/bioinformatics/btw351. PMID: 27357171; PMCID: PMC5048064</p>
<p>3. Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for Survival Data with High-Dimensional Mediators.
Bioinformatics. 2021. DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID: PMC8570823</p>
<p>4. Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation Effect Selection in High-dimensional and Compositional Microbiome data.
Stat Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID: PMC7855955</p>
<p>5. Zhang H, Chen J, Li Z, Liu L. Testing for Mediation Effect with Application to Human Microbiome Data.
Stat Biosci. 2021. DOI: 10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450</p>
<p>6. Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L. HIMA2: High-dimensional Mediation Analysis and Its Application in
Epigenome-wide DNA Methylation Data. BMC Bioinformatics. 2022. DOI: 10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002</p>
<p>7. Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L. High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI: 10.1093/bioinformatics/btae055. PMID: 38290773; PMCID: PMC10873903</p>
<p>8. Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI: 10.1007/s12561-024-09447-4.</p>
<p>9. Liu L, Zhang H, Zheng Y, Gao T, Zheng C, Zhang K, Hou L, Liu L. High-dimensional mediation analysis for longitudinal mediators and
survival outcomes. Briefings in Bioinformatics. 2025. DOI: 10.1093/bib/bbaf206. PMID: 40350699 PMCID: PMC12066418</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Note: In the following examples, M1, M2, and M3 are true mediators.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 1 (continuous outcome - linear HIMA):</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ContinuousOutcome</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">PhenoData</span></span></span>
<span class="r-in"><span><span class="va">mediator_data</span> <span class="op">&lt;-</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">Mediator</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">e1</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>, <span class="co"># Can be "DBlasso" for hima_dblasso</span></span></span>
<span class="r-in"><span>  scale <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disabled only for simulation data</span></span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e1</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Efficient HIMA (only applicable to mediators and outcomes that are</span></span></span>
<span class="r-in"><span><span class="co"># both continuous and normally distributed.)</span></span></span>
<span class="r-in"><span><span class="va">e1e</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>  efficient <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>, <span class="co"># Efficient HIMA does not support DBlasso</span></span></span>
<span class="r-in"><span>  scale <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disabled only for simulation data</span></span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e1e</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 2 (binary outcome - logistic HIMA):</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">BinaryOutcome</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">BinaryOutcome</span><span class="op">$</span><span class="va">PhenoData</span></span></span>
<span class="r-in"><span><span class="va">mediator_data</span> <span class="op">&lt;-</span> <span class="va">BinaryOutcome</span><span class="op">$</span><span class="va">Mediator</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">e2</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="va">Disease</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span></span>
<span class="r-in"><span>  scale <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disabled only for simulation data</span></span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 3 (time-to-event outcome - survival HIMA):</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SurvivalData</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">SurvivalData</span><span class="op">$</span><span class="va">PhenoData</span></span></span>
<span class="r-in"><span><span class="va">mediator_data</span> <span class="op">&lt;-</span> <span class="va">SurvivalData</span><span class="op">$</span><span class="va">Mediator</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">e3</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="fu">Surv</span><span class="op">(</span><span class="va">Time</span>, <span class="va">Status</span><span class="op">)</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span>,</span></span>
<span class="r-in"><span>  scale <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disabled only for simulation data</span></span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="co"># Parallel computing feature is recommended</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e3</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Longitudinal mediator + survival HIMA:</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SurvivalLongData</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">SurvivalLongData</span><span class="op">$</span><span class="va">PhenoData</span></span></span>
<span class="r-in"><span><span class="va">mediator_data</span> <span class="op">&lt;-</span> <span class="va">SurvivalLongData</span><span class="op">$</span><span class="va">Mediator</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">e3long</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="fu">Surv</span><span class="op">(</span><span class="va">Tstart</span>, <span class="va">Tstop</span>, <span class="va">Status</span><span class="op">)</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"lasso"</span>,</span></span>
<span class="r-in"><span>  longitudinal <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  id.var <span class="op">=</span> <span class="st">"ID"</span>,</span></span>
<span class="r-in"><span>  scale <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disabled only for simulation data</span></span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="co"># Parallel computing feature is recommended</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e3long</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Example 4 (compositional data as mediator, e.g., microbiome):</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MicrobiomeData</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">MicrobiomeData</span><span class="op">$</span><span class="va">PhenoData</span></span></span>
<span class="r-in"><span><span class="va">mediator_data</span> <span class="op">&lt;-</span> <span class="va">MicrobiomeData</span><span class="op">$</span><span class="va">Mediator</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">e4</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"compositional"</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="co"># Scaling is always enabled internally for hima_microbiome</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e4</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#' # Example 5 (quantile mediation analysis - quantile HIMA):</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">QuantileData</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pheno_data</span> <span class="op">&lt;-</span> <span class="va">QuantileData</span><span class="op">$</span><span class="va">PhenoData</span></span></span>
<span class="r-in"><span><span class="va">mediator_data</span> <span class="op">&lt;-</span> <span class="va">QuantileData</span><span class="op">$</span><span class="va">Mediator</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Note that the function will prompt input for quantile level.</span></span></span>
<span class="r-in"><span><span class="va">e5</span> <span class="op">&lt;-</span> <span class="fu">hima</span><span class="op">(</span><span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span></span>
<span class="r-in"><span>  data.pheno <span class="op">=</span> <span class="va">pheno_data</span>,</span></span>
<span class="r-in"><span>  data.M <span class="op">=</span> <span class="va">mediator_data</span>,</span></span>
<span class="r-in"><span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span></span>
<span class="r-in"><span>  quantile <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>, <span class="co"># Quantile HIMA does not support DBlasso</span></span></span>
<span class="r-in"><span>  scale <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Disabled only for simulation data</span></span></span>
<span class="r-in"><span>  tau <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.7</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="co"># Specify multiple quantile level</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">e5</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yinan Zheng, Haixiang Zhang, Lifang Hou, Lei Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

