<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>High-Dimensional Mediation Analysis • HIMA</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="High-Dimensional Mediation Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HIMA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/hima-vignette.html">Tutorials</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/YinanZheng/HIMA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>High-Dimensional Mediation Analysis</h1>
            <h3 data-toc-skip class="subtitle">A Guide to Using the HIMA
Package</h3>
                        <h4 data-toc-skip class="author">The HIMA
Development Team</h4>
            
            <h4 data-toc-skip class="date">2025-01-25</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/YinanZheng/HIMA/blob/HEAD/vignettes/hima-vignette.Rmd"><code>vignettes/hima-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>hima-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Mediation analysis is a statistical method used to explore the
mechanisms by which an independent variable influences a dependent
variable through one or more intermediary variables, known as mediators.
This process involves assessing the indirect, direct, and total effects
within a defined statistical framework. The primary goal of mediation
analysis is to test the mediation hypothesis, which posits that the
effect of an independent variable on a dependent variable is partially
or fully mediated by intermediate variables. By examining these
pathways, researchers can gain a deeper understanding of the causal
relationships among variables. Mediation analysis is particularly
valuable in refining interventions to enhance their effectiveness in
clinical trials and in observational studies, where it helps identify
key intervention targets and elucidate the mechanisms underlying complex
diseases.</p>
</div>
<div class="section level2">
<h2 id="package-overview">Package Overview<a class="anchor" aria-label="anchor" href="#package-overview"></a>
</h2>
<p>The <code>HIMA</code> package provides robust tools for estimating
and testing high-dimensional mediation effects, specifically designed
for modern <code>omic</code> data, including epigenetics,
transcriptomics, and microbiomics. It supports high-dimensional
mediation analysis across continuous, binary, and survival outcomes,
with specialized methods tailored for compositional microbiome data and
quantile mediation analysis.</p>
<p>At the core of the package is the <code>hima</code> function, a
flexible and powerful wrapper that integrates various high-dimensional
mediation analysis methods for both effect estimation and hypothesis
testing. The <code>hima</code> function automatically selects the most
suitable analytical approach based on the outcome and mediator data
type, streamlining complex workflows and reducing user burden.</p>
<p><code>hima</code> is designed with extensibility in mind, allowing
seamless integration of future features and methods. This ensures a
consistent, user-friendly interface while staying adaptable to evolving
analytical needs.</p>
</div>
<div class="section level2">
<h2 id="data-preparation-and-settings">Data Preparation and Settings<a class="anchor" aria-label="anchor" href="#data-preparation-and-settings"></a>
</h2>
<p>The <code>hima</code> function provides a flexible and user-friendly
interface for performing high-dimensional mediation analysis. It
supports a variety of analysis methods tailored for continuous, binary,
survival, and compositional data. Below is an overview of the
<code>hima</code> function and its key parameters:</p>
<div class="section level4">
<h4 id="hima-function-interface">
<code>hima</code> Function Interface<a class="anchor" aria-label="anchor" href="#hima-function-interface"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">formula</span>,          <span class="co"># The model formula specifying outcome, exposure, and covariate(s)</span></span>
<span>  <span class="va">data.pheno</span>,       <span class="co"># Data frame with outcome, exposure, and covariate(s)</span></span>
<span>  <span class="va">data.M</span>,           <span class="co"># Data frame or matrix of high-dimensional mediators</span></span>
<span>  <span class="va">mediator.type</span>,    <span class="co"># Type of mediators: "gaussian", "negbin", or "compositional"</span></span>
<span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span>,  <span class="co"># Penalty method: "DBlasso", "MCP", "SCAD", or "lasso"</span></span>
<span>  quantile <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># Use quantile mediation analysis (default: FALSE)</span></span>
<span>  efficient <span class="op">=</span> <span class="cn">FALSE</span>,<span class="co"># Use efficient mediation analysis (default: FALSE)</span></span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span>,     <span class="co"># Scale data (default: TRUE)</span></span>
<span>  sigcut <span class="op">=</span> <span class="fl">0.05</span>,    <span class="co"># Significance cutoff for mediator selection</span></span>
<span>  contrast <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># Named list of contrasts for factor covariate(s)</span></span>
<span>  subset <span class="op">=</span> <span class="cn">NULL</span>,    <span class="co"># Optional subset of observations</span></span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>   <span class="co"># Display progress messages (default: FALSE)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>To use the <code>hima</code> function, ensure your data is prepared
according to the following guidelines:</p>
</div>
<div class="section level4">
<h4 id="formula-argument-formula">1. Formula Argument (<code>formula</code>)<a class="anchor" aria-label="anchor" href="#formula-argument-formula"></a>
</h4>
<p>Define the model formula to specify the relationship between the
<code>Outcome</code>, <code>Exposure</code>, and
<code>Covariate(s)</code>. Ensure the following:</p>
<ul>
<li><p><strong>General Form</strong>: Use the format
<code>Outcome ~ Exposure + Covariate(s)</code>. Note that the
<code>Exposure</code> variable represents the exposure of interest
(e.g., “Treatment” in the demo examples) and it has to be listed as the
first independent variable in the formula. <code>Covariate(s)</code> are
optional.</p></li>
<li><p><strong>Survival Data</strong>: For survival analysis, use the
format <code>Surv(time, event) ~ Exposure + Covariate(s)</code>. See
data examples <code>SurvivalData$PhenoData</code> for more
details.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="phenotype-data-data-pheno">2. Phenotype Data (<code>data.pheno</code>)<a class="anchor" aria-label="anchor" href="#phenotype-data-data-pheno"></a>
</h4>
<p>The <code>data.pheno</code> object should be a
<code>data.frame</code> or <code>matrix</code> containing the phenotype
information for the analysis (without missing values). Key requirements
include:</p>
<ul>
<li><p><strong>Rows:</strong> Represent samples.</p></li>
<li><p><strong>Columns:</strong> Include variables such as the outcome,
treatment, and optional covariate(s).</p></li>
<li><p><strong>Formula Consistency:</strong> Ensure that all variables
specified in the <code>formula</code> argument (e.g.,
<code>Outcome</code>, <code>Treatment</code>, and
<code>Covariate(s)</code>) are present in
<code>data.pheno</code>.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="mediator-data-data-m">3. Mediator Data (<code>data.M</code>)<a class="anchor" aria-label="anchor" href="#mediator-data-data-m"></a>
</h4>
<p>The <code>data.M</code> object should be a <code>data.frame</code> or
<code>matrix</code> containing high-dimensional mediators (without
missing values). Key requirements include:</p>
<ul>
<li><p><strong>Rows:</strong> Represent samples, aligned with the rows
in <code>data.pheno</code>.</p></li>
<li><p><strong>Columns:</strong> Represent mediators (e.g., CpGs, genes,
or other molecular features).</p></li>
<li>
<p><strong>Mediator Type:</strong> Specify the type of mediators in
the <code>mediator.type</code> argument. Supported types include:</p>
<ul>
<li>
<code>"gaussian"</code> for continuous mediators (default, e.g., DNA
methylation data).</li>
<li>
<code>"negbin"</code> for count data (e.g., transcriptomic
data).</li>
<li>
<code>"compositional"</code> for microbiome or other compositional
data.</li>
</ul>
</li>
</ul>
</div>
<div class="section level4">
<h4 id="about-data-scaling">4. About data scaling<a class="anchor" aria-label="anchor" href="#about-data-scaling"></a>
</h4>
<p>In most real-world data analysis scenarios, <code>scale</code> is
typically set to <code>TRUE</code>, ensuring that the exposure (variable
of interest), mediators, and covariate(s) (if included) are standardized
to a mean of zero and a variance of one. No scaling will be applied to
<code>Outcome</code>. However, if your data is already
pre-standardized—such as in simulation studies or when using our demo
dataset-<code>scale</code> should be set to <code>FALSE</code> to
prevent introducing biases or altering the original data structure.</p>
<p>When applying <code>HIMA</code> to simulated data, if
<code>scale</code> is set to <code>TRUE</code>, it is imperative to
preprocess the mediators by scaling them to have a mean of zero and a
variance of one prior to generating the outcome variables.</p>
</div>
</div>
<div class="section level2">
<h2 id="applications-and-examples">Applications and Examples<a class="anchor" aria-label="anchor" href="#applications-and-examples"></a>
</h2>
<div class="section level3">
<h3 id="load-the-hima-package">Load the <code>HIMA</code> Package<a class="anchor" aria-label="anchor" href="#load-the-hima-package"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YinanZheng/HIMA/">HIMA</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="continuous-outcome-analysis">Continuous Outcome Analysis<a class="anchor" aria-label="anchor" href="#continuous-outcome-analysis"></a>
</h3>
<p>When analyzing continuous and normally distributed outcomes and
mediators, we can use the following code snippet:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ContinuousOutcome</span><span class="op">)</span></span>
<span><span class="va">hima_continuous.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Demo data is already standardized</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_continuous.fit</span>, desc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="co"># `desc = TRUE` option to show the description of the output results</span></span></code></pre></div>
<p>The <code>penaly</code> can be set to <code>DBlasso</code> which may
help detect more mediators with weaker signals.</p>
</div>
<div class="section level3">
<h3 id="efficient-hima">Efficient HIMA<a class="anchor" aria-label="anchor" href="#efficient-hima"></a>
</h3>
<p>For continuous and normally distributed mediators and outcomes, an
efficient HIMA method can be activated with the
<code>efficient = TRUE</code> option (penalty should be <code>MCP</code>
for the best results). This method may also provide greater statistical
power to detect mediators with weaker signals.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hima_efficient.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  efficient <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"lasso"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Demo data is already standardized</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_efficient.fit</span>, desc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="co"># Note that the efficient HIMA is controlling FDR</span></span></code></pre></div>
<p>It is recommended to try different <code>penalty</code> options and
<code>efficient</code> option to find the best one for your data.</p>
</div>
<div class="section level3">
<h3 id="binary-outcome-analysis">Binary Outcome Analysis<a class="anchor" aria-label="anchor" href="#binary-outcome-analysis"></a>
</h3>
<p>The package can handle binary outcomes based on logistic
regression:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">BinaryOutcome</span><span class="op">)</span></span>
<span><span class="va">hima_binary.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Disease</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">BinaryOutcome</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">BinaryOutcome</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Demo data is already standardized</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_binary.fit</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="survival-outcome-analysis">Survival Outcome Analysis<a class="anchor" aria-label="anchor" href="#survival-outcome-analysis"></a>
</h3>
<p>For survival data, <code>HIMA</code> incorporates a Cox proportional
hazards approach. Here is an example of survival outcome analysis using
<code>HIMA</code>:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SurvivalData</span><span class="op">)</span></span>
<span><span class="va">hima_survival.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="fu">Surv</span><span class="op">(</span><span class="va">Time</span>, <span class="va">Status</span><span class="op">)</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">SurvivalData</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">SurvivalData</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Demo data is already standardized</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_survival.fit</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="microbiome-mediation-analysis">Microbiome Mediation Analysis<a class="anchor" aria-label="anchor" href="#microbiome-mediation-analysis"></a>
</h3>
<p>For compositional microbiome data, <code>HIMA</code> employs
isometric Log-Ratio transformations. Here is an example of microbiome
mediation analysis using <code>HIMA</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MicrobiomeData</span><span class="op">)</span></span>
<span><span class="va">hima_microbiome.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">MicrobiomeData</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">MicrobiomeData</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"compositional"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_microbiome.fit</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quantile-mediation-analysis">Quantile Mediation Analysis<a class="anchor" aria-label="anchor" href="#quantile-mediation-analysis"></a>
</h3>
<p>Perform quantile mediation analysis using the
<code>quantile = TRUE</code> option and specify <code>tau</code> for
desired quantile(s):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">QuantileData</span><span class="op">)</span></span>
<span><span class="va">hima_quantile.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">QuantileData</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">QuantileData</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  quantile <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  tau <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.7</span><span class="op">)</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># Demo data is already standardized</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_quantile.fit</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W,
Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, Lv J, Baccarelli A,
Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects
in Epigenetic Studies. Bioinformatics. 2016. DOI:
10.1093/bioinformatics/btw351. PMID: 27357171; PMCID:
PMC5048064</p></li>
<li><p>Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for
Survival Data with High-Dimensional Mediators. Bioinformatics. 2021.
DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID:
PMC8570823</p></li>
<li><p>Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation Effect
Selection in High-dimensional and Compositional Microbiome data. Stat
Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID:
PMC7855955</p></li>
<li><p>Zhang H, Chen J, Li Z, Liu L. Testing for Mediation Effect with
Application to Human Microbiome Data. Stat Biosci. 2021. DOI:
10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450</p></li>
<li><p>Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L.
HIMA2: High-dimensional Mediation Analysis and Its Application in
Epigenome-wide DNA Methylation Data. BMC Bioinformatics. 2022. DOI:
10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002</p></li>
<li><p>Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L.
High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI:
10.1093/bioinformatics/btae055. PMID: 38290773; PMCID:
PMC10873903</p></li>
<li><p>Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient
Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI:
10.1007/s12561-024-09447-4.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yinan Zheng, Haixiang Zhang, Lifang Hou, Lei Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
