<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>High-Dimensional Mediation Analysis • HIMA</title>
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="High-Dimensional Mediation Analysis">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HIMA</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/hima-vignette.html">Tutorials</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">News</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="nav-link" href="https://github.com/YinanZheng/HIMA/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>High-Dimensional Mediation Analysis</h1>
            <h3 data-toc-skip class="subtitle">A Guide to Using the HIMA
Package</h3>
                        <h4 data-toc-skip class="author">The HIMA
Development Team</h4>
            
            <h4 data-toc-skip class="date">2025-01-25</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/YinanZheng/HIMA/blob/HEAD/vignettes/hima-vignette.Rmd"><code>vignettes/hima-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>hima-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Mediation analysis is a statistical methodology employed to
investigate the underlying mechanisms through which an independent
variable influences a dependent variable through an intermediary
variable (mediator). This approach typically involves evaluating
indirect, direct, and total effects within a specified statistical
framework. A key objective of mediation analysis is to test the
mediation hypothesis, which suggests that the impact of the independent
variable on the dependent variable is transmitted through one or more
mediating variables. By investigating these pathways, we can achieve a
more comprehensive understanding of the causal relationships among the
variables. For instance, mediation analysis can enhance the refinement
of interventions to improve their efficacy in clinical trials. In
observational studies, mediation analysis can serve to pinpoint
intervention targets and elucidate the underlying mechanisms of
diseases.</p>
</div>
<div class="section level2">
<h2 id="package-overview">Package Overview<a class="anchor" aria-label="anchor" href="#package-overview"></a>
</h2>
<p>The <code>HIMA</code> package provides tools for estimating and
testing high-dimensional mediation effects, tailored for modern
<code>omic</code> data such as epigenetics, transcriptomics, and
microbiomics. The function, <code>hima</code>, serves as a comprehensive
wrapper function designed to implement various high-dimensional
mediation analysis methods for both estimating and testing mediation
effects. <code>hima</code> can automatically select the appropriate
method based on the outcome and mediator data type. The
<code>hima</code> function is designed to be extensible, allowing future
features and methods to seamlessly integrate within its framework,
ensuring a consistent and user-friendly experience.</p>
<div class="section level4">
<h4 id="key-features">Key Features<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h4>
<ul>
<li>High-dimensional mediation analysis for continuous, binary, and
survival outcomes.</li>
<li>Specialized methods for compositional microbiome and quantile
mediation analysis.</li>
<li>Flexible penalty options including <code>DBlasso</code>,
<code>MCP</code>, and <code>SCAD</code>.</li>
<li>High-level wrapper function <code>hima</code> for easy
implementation.</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="package-installation">Package Installation<a class="anchor" aria-label="anchor" href="#package-installation"></a>
</h2>
<p>HIMA is now available in the R CRAN repository. To install from
CRAN:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"HIMA"</span><span class="op">)</span></span></code></pre></div>
<p>To install the development version from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"yinanzheng/HIMA"</span><span class="op">)</span></span></code></pre></div>
<p>If the package “qvalue” is not found, please first install the
“qvalue” package through Bioconductor:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"qvalue"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-preparation-and-settings">Data Preparation and Settings<a class="anchor" aria-label="anchor" href="#data-preparation-and-settings"></a>
</h2>
<p>To use the <code>hima</code> function, ensure your data is prepared
according to the following guidelines:</p>
<div class="section level4">
<h4 id="formula-argument-formula">1. Formula Argument (<code>formula</code>)<a class="anchor" aria-label="anchor" href="#formula-argument-formula"></a>
</h4>
<p>Define the model formula to specify the relationship between the
outcome, treatment, and covariates. Ensure the following: -
<strong>General Form</strong>: Use the format
<code>Outcome ~ Exposure + Covariates</code>. Note that the
<code>Exposure</code> variable represents the exposure of interest
(e.g., “Treatment” in the demo examples) and it has to be listed as the
first independent variable in the formula. <code>Covariates</code> are
optional. - <strong>Interaction Terms</strong>: Optionally include
interaction terms if required by your analysis. - <strong>Survival
Data</strong>: For survival analysis, use the format
<code>Surv(time, event) ~ Exposure + Covariates</code>. See data
examples <code>SurvivalData$PhenoData</code> for more details.</p>
</div>
<div class="section level4">
<h4 id="phenotype-data-data-pheno">2. Phenotype Data (<code>data.pheno</code>)<a class="anchor" aria-label="anchor" href="#phenotype-data-data-pheno"></a>
</h4>
<p>The <code>data.pheno</code> object should be a
<code>data.frame</code> or <code>matrix</code> containing the phenotype
information for the analysis. Key requirements include: -
<strong>Rows:</strong> Represent samples. - <strong>Columns:</strong>
Include variables such as the outcome, treatment, and optional
covariates. - <strong>Formula Consistency:</strong> Ensure that all
variables specified in the <code>formula</code> argument (e.g.,
<code>Outcome</code>, <code>Treatment</code>, and
<code>Covariates</code>) are present in <code>data.pheno</code>.</p>
</div>
<div class="section level4">
<h4 id="mediator-data-data-m">3. Mediator Data (<code>data.M</code>)<a class="anchor" aria-label="anchor" href="#mediator-data-data-m"></a>
</h4>
<p>The <code>data.M</code> object should be a <code>data.frame</code> or
<code>matrix</code> containing high-dimensional mediators. Key
requirements include: - <strong>Rows:</strong> Represent samples,
aligned with the rows in <code>data.pheno</code>. -
<strong>Columns:</strong> Represent mediators (e.g., CpGs, genes, or
other molecular features). - <strong>Mediator Type:</strong> Specify the
type of mediators in the <code>mediator.type</code> argument. Supported
types include (more supporting types will be added in the future): -
<code>"gaussian"</code> for continuous mediators (default, e.g., DNA
methylation data). - <code>"negbin"</code> for count data (e.g.,
transcriptomic data). - <code>"compositional"</code> for microbiome or
other compositional data.</p>
</div>
<div class="section level4">
<h4 id="about-data-scaling">4. About data scaling<a class="anchor" aria-label="anchor" href="#about-data-scaling"></a>
</h4>
<p>In most real-world data analysis scenarios, <code>scale</code> is
typically set to <code>TRUE</code>, ensuring that the exposure (variable
of interest), mediators, and covariates (if included) are standardized
to a mean of zero and a variance of one. However, if your data is
already pre-standardized—such as in simulation studies or when using our
demo dataset-<code>scale</code> should be set to <code>FALSE</code> to
prevent introducing biases or altering the original data structure. When
applying <code>HIMA</code> to simulated data, if <code>scale</code> is
set to <code>TRUE</code>, it is imperative to preprocess the mediators
by scaling them to have a mean of zero and a variance of one prior to
generating the outcome variables.</p>
</div>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<div class="section level3">
<h3 id="load-the-hima-package">Load the <code>HIMA</code> Package<a class="anchor" aria-label="anchor" href="#load-the-hima-package"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/YinanZheng/HIMA/">HIMA</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="continuous-outcome-analysis">Continuous Outcome Analysis<a class="anchor" aria-label="anchor" href="#continuous-outcome-analysis"></a>
</h3>
<p>We begin with an example of analyzing continuous outcomes using the
<code>HIMA</code> package. The following code snippet demonstrates how
to perform a high-dimensional mediation analysis with continuous
outcomes:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">ContinuousOutcome</span><span class="op">)</span></span>
<span><span class="va">hima_continuous.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Running linear HIMA with MCP penalty...</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening ...     (3:38:50 AM)</span></span>
<span><span class="co">#&gt; Step 2: Penalized estimate (MCP) ...     (3:38:52 AM)</span></span>
<span><span class="co">#&gt; Step 3: Joint significance test ...     (3:38:52 AM)</span></span>
<span><span class="co">#&gt; Done!     (3:38:52 AM)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_continuous.fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID      alpha      beta alpha*beta Relative Importance (%)      p-value</span></span>
<span><span class="co">#&gt; 1 M3 -0.3041928 -1.249628  0.3801277                39.51341 2.049845e-12</span></span>
<span><span class="co">#&gt; 2 M2  0.2659784  1.371307  0.3647379                37.91367 1.203775e-09</span></span>
<span><span class="co">#&gt; 3 M1  0.1448829  1.498841  0.2171565                22.57292 1.109704e-03</span></span></code></pre></div>
<p>We can use the <code>summary</code> function with the
<code>desc = TRUE</code> option to show the description of the output
results:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_continuous.fit</span>, desc<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID      alpha      beta alpha*beta Relative Importance (%)      p-value</span></span>
<span><span class="co">#&gt; 1 M3 -0.3041928 -1.249628  0.3801277                39.51341 2.049845e-12</span></span>
<span><span class="co">#&gt; 2 M2  0.2659784  1.371307  0.3647379                37.91367 1.203775e-09</span></span>
<span><span class="co">#&gt; 3 M1  0.1448829  1.498841  0.2171565                22.57292 1.109704e-03</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Variable Descriptions:</span></span>
<span><span class="co">#&gt; ID: Mediator ID</span></span>
<span><span class="co">#&gt; alpha: Effect of exposure on mediator</span></span>
<span><span class="co">#&gt; beta: Effect of mediator on outcome</span></span>
<span><span class="co">#&gt; alpha*beta: Mediation (indirect) effect</span></span>
<span><span class="co">#&gt; Relative Importance (%): Relative importance of the mediator </span></span>
<span><span class="co">#&gt;   out of all significant mediators</span></span>
<span><span class="co">#&gt; p-value: Joint raw p-value of significant mediator </span></span>
<span><span class="co">#&gt;   selected based on Bonferroni-adjusted p &lt; 0.05</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="efficient-hima">Efficient HIMA<a class="anchor" aria-label="anchor" href="#efficient-hima"></a>
</h3>
<p>For continuous mediators and outcomes, use the efficient HIMA method
with the <code>efficient = TRUE</code> option. This method is
computationally efficient for large datasets and high-dimensional
mediators. For example, consider the following code snippet for an
efficient HIMA analysis:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hima_efficient.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">ContinuousOutcome</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  efficient <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Running efficient HIMA with MCP penalty...</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening + minimax concave penalty (MCP) ...     (3:38:52 AM)</span></span>
<span><span class="co">#&gt; Step 2: Refitted partial regression ...     (3:38:52 AM)</span></span>
<span><span class="co">#&gt; Step 3: Divide-aggregate composite-null test (DACT) ...     (3:38:53 AM)</span></span>
<span><span class="co">#&gt; Done!     (3:38:53 AM)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_efficient.fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID      alpha      beta alpha*beta Relative Importance (%)      p-value</span></span>
<span><span class="co">#&gt; 1 M3 -0.3041928 -1.249628  0.3801277                39.51341 3.311590e-15</span></span>
<span><span class="co">#&gt; 2 M2  0.2659784  1.371307  0.3647379                37.91367 4.874037e-12</span></span>
<span><span class="co">#&gt; 3 M1  0.1448829  1.498841  0.2171565                22.57292 1.865340e-05</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="binary-outcome-analysis">Binary Outcome Analysis<a class="anchor" aria-label="anchor" href="#binary-outcome-analysis"></a>
</h3>
<p>The package can also handle binary outcomes. Here is an example of
binary outcome analysis using <code>HIMA</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">BinaryOutcome</span><span class="op">)</span></span>
<span><span class="va">hima_binary.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Disease</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">BinaryOutcome</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">BinaryOutcome</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Running logistic HIMA with MCP penalty...</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening ...     (3:38:53 AM)</span></span>
<span><span class="co">#&gt; Step 2: Penalized estimate (MCP) ...     (3:38:55 AM)</span></span>
<span><span class="co">#&gt;     Using alpha estimated in Step 1 ...   (3:38:55 AM)</span></span>
<span><span class="co">#&gt; Step 3: Joint significance test ...     (3:38:55 AM)</span></span>
<span><span class="co">#&gt; Done!     (3:38:55 AM)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_binary.fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID      alpha      beta alpha*beta Relative Importance (%)      p-value</span></span>
<span><span class="co">#&gt; 1 M3 -0.2638030 -1.336908  0.3526802                38.63434 1.610871e-09</span></span>
<span><span class="co">#&gt; 2 M1  0.2490828  1.183759  0.2948540                32.29977 8.570595e-09</span></span>
<span><span class="co">#&gt; 3 M2  0.2446872  1.084376  0.2653330                29.06589 2.984815e-08</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="survival-outcome-analysis">Survival Outcome Analysis<a class="anchor" aria-label="anchor" href="#survival-outcome-analysis"></a>
</h3>
<p>For survival data, <code>HIMA</code> incorporates a Cox proportional
hazards approach. Here is an example of survival outcome analysis using
<code>HIMA</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">SurvivalData</span><span class="op">)</span></span>
<span><span class="va">hima_survival.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="fu">Surv</span><span class="op">(</span><span class="va">Time</span>, <span class="va">Status</span><span class="op">)</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">SurvivalData</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">SurvivalData</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Running survival HIMA with de-biased Lasso penalty...</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening ...     (3:38:55 AM)</span></span>
<span><span class="co">#&gt; Step 2: De-biased Lasso estimates ...     (3:38:55 AM)</span></span>
<span><span class="co">#&gt; Step 3: Multiple-testing procedure ...     (3:39:21 AM)</span></span>
<span><span class="co">#&gt; Done!     (3:39:21 AM)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_survival.fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID      alpha       beta alpha*beta Relative Importance (%)      p-value</span></span>
<span><span class="co">#&gt; 1 M2  0.7630170  0.7480588  0.5707815                26.44027 3.365552e-11</span></span>
<span><span class="co">#&gt; 2 M3 -0.8212199 -0.9569047  0.7858292                36.40191 4.325473e-11</span></span>
<span><span class="co">#&gt; 3 M1  0.8215721  0.9763566  0.8021474                37.15782 1.134386e-10</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="microbiome-mediation-analysis">Microbiome Mediation Analysis<a class="anchor" aria-label="anchor" href="#microbiome-mediation-analysis"></a>
</h3>
<p>For compositional microbiome data, <code>HIMA</code> employs
specialized transformations. Here is an example of microbiome mediation
analysis using <code>HIMA</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MicrobiomeData</span><span class="op">)</span></span>
<span><span class="va">hima_microbiome.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">MicrobiomeData</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">MicrobiomeData</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"compositional"</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"DBlasso"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Running compositional HIMA with de-biased Lasso penalty...</span></span>
<span><span class="co">#&gt; Step 1: ILR Transformation and De-biased Lasso estimates ...  (3:39:21 AM)</span></span>
<span><span class="co">#&gt; Step 2: Closted testing-based procedure ...     (3:40:14 AM)</span></span>
<span><span class="co">#&gt; Done!     (3:40:14 AM)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_microbiome.fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 3 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID     alpha      beta alpha*beta Relative Importance (%) p-value</span></span>
<span><span class="co">#&gt; 1 M1 0.8161109  3.910193   3.191151                50.11182       0</span></span>
<span><span class="co">#&gt; 2 M2 0.8171704 -2.250231  -1.838822                28.87570       0</span></span>
<span><span class="co">#&gt; 3 M3 0.7920133 -1.689476  -1.338088                21.01248       0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quantile-mediation-analysis">Quantile Mediation Analysis<a class="anchor" aria-label="anchor" href="#quantile-mediation-analysis"></a>
</h3>
<p>Perform quantile mediation analysis using the
<code>quantile = TRUE</code> option. Here is an example of quantile
mediation analysis using <code>HIMA</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">QuantileData</span><span class="op">)</span></span>
<span><span class="va">hima_quantile.fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hima.html">hima</a></span><span class="op">(</span></span>
<span>  <span class="va">Outcome</span> <span class="op">~</span> <span class="va">Treatment</span> <span class="op">+</span> <span class="va">Sex</span> <span class="op">+</span> <span class="va">Age</span>,</span>
<span>  data.pheno <span class="op">=</span> <span class="va">QuantileData</span><span class="op">$</span><span class="va">PhenoData</span>,</span>
<span>  data.M <span class="op">=</span> <span class="va">QuantileData</span><span class="op">$</span><span class="va">Mediator</span>,</span>
<span>  mediator.type <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>  quantile <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  penalty <span class="op">=</span> <span class="st">"MCP"</span>,</span>
<span>  tau <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.5</span>, <span class="fl">0.7</span><span class="op">)</span>,</span>
<span>  scale <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Running quantile HIMA with MCP penalty...</span></span>
<span><span class="co">#&gt; Running penalized quantile regression with tau = 0.3 ...     (3:40:14 AM)</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening ...     (3:40:14 AM)</span></span>
<span><span class="co">#&gt; Step 2: Penalized estimate (MCP) ...     (3:40:14 AM)</span></span>
<span><span class="co">#&gt; Step 3: Joint significance test ...     (3:40:16 AM)</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; Running penalized quantile regression with tau = 0.5 ...     (3:40:18 AM)</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening ...     (3:40:18 AM)</span></span>
<span><span class="co">#&gt; Step 2: Penalized estimate (MCP) ...     (3:40:18 AM)</span></span>
<span><span class="co">#&gt; Step 3: Joint significance test ...     (3:40:20 AM)</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; Running penalized quantile regression with tau = 0.7 ...     (3:40:21 AM)</span></span>
<span><span class="co">#&gt; Step 1: Sure Independent Screening ...     (3:40:21 AM)</span></span>
<span><span class="co">#&gt; Step 2: Penalized estimate (MCP) ...     (3:40:22 AM)</span></span>
<span><span class="co">#&gt; Step 3: Joint significance test ...     (3:40:24 AM)</span></span>
<span><span class="co">#&gt;  </span></span>
<span><span class="co">#&gt; Done!     (3:40:26 AM)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">hima_quantile.fit</span><span class="op">)</span></span>
<span><span class="co">#&gt; Summary of HIMA results:</span></span>
<span><span class="co">#&gt; -----------------------------------</span></span>
<span><span class="co">#&gt; Number of significant mediators: 9 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Top mediators (sorted by p-value):</span></span>
<span><span class="co">#&gt;   ID      alpha      beta alpha*beta Relative Importance (%)      p-value tau</span></span>
<span><span class="co">#&gt; 1 M1  0.8061352  1.755566   1.415224                35.46531 0.000000e+00 0.3</span></span>
<span><span class="co">#&gt; 2 M1  0.8061352  1.555867   1.254239                32.24338 0.000000e+00 0.3</span></span>
<span><span class="co">#&gt; 3 M2  0.8485185  1.603046   1.360214                34.96773 0.000000e+00 0.3</span></span>
<span><span class="co">#&gt; 4 M3 -0.8608720 -1.481589   1.275458                32.78888 0.000000e+00 0.5</span></span>
<span><span class="co">#&gt; 5 M1  0.8061352  1.597611   1.287890                31.38334 0.000000e+00 0.5</span></span>
<span><span class="co">#&gt; 6 M2  0.8485185  1.581736   1.342132                32.70512 0.000000e+00 0.5</span></span>
<span><span class="co">#&gt; 7 M3 -0.8608720 -1.711887   1.473715                35.91154 0.000000e+00 0.7</span></span>
<span><span class="co">#&gt; 8 M2  0.8485185  1.616386   1.371534                34.37044 2.318146e-13 0.7</span></span>
<span><span class="co">#&gt; 9 M3 -0.8608720 -1.398219   1.203688                30.16424 2.896883e-10 0.7</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Zhang H, Zheng Y, Zhang Z, Gao T, Joyce B, Yoon G, Zhang W,
Schwartz J, Just A, Colicino E, Vokonas P, Zhao L, Lv J, Baccarelli A,
Hou L, Liu L. Estimating and Testing High-dimensional Mediation Effects
in Epigenetic Studies. Bioinformatics. 2016. DOI:
10.1093/bioinformatics/btw351. PMID: 27357171; PMCID:
PMC5048064</p></li>
<li><p>Zhang H, Zheng Y, Hou L, Zheng C, Liu L. Mediation Analysis for
Survival Data with High-Dimensional Mediators. Bioinformatics. 2021.
DOI: 10.1093/bioinformatics/btab564. PMID: 34343267; PMCID:
PMC8570823</p></li>
<li><p>Zhang H, Chen J, Feng Y, Wang C, Li H, Liu L. Mediation Effect
Selection in High-dimensional and Compositional Microbiome data. Stat
Med. 2021. DOI: 10.1002/sim.8808. PMID: 33205470; PMCID:
PMC7855955</p></li>
<li><p>Zhang H, Chen J, Li Z, Liu L. Testing for Mediation Effect with
Application to Human Microbiome Data. Stat Biosci. 2021. DOI:
10.1007/s12561-019-09253-3. PMID: 34093887; PMCID: PMC8177450</p></li>
<li><p>Perera C, Zhang H, Zheng Y, Hou L, Qu A, Zheng C, Xie K, Liu L.
HIMA2: High-dimensional Mediation Analysis and Its Application in
Epigenome-wide DNA Methylation Data. BMC Bioinformatics. 2022. DOI:
10.1186/s12859-022-04748-1. PMID: 35879655; PMCID: PMC9310002</p></li>
<li><p>Zhang H, Hong X, Zheng Y, Hou L, Zheng C, Wang X, Liu L.
High-Dimensional Quantile Mediation Analysis with Application to a Birth
Cohort Study of Mother–Newborn Pairs. Bioinformatics. 2024. DOI:
10.1093/bioinformatics/btae055. PMID: 38290773; PMCID:
PMC10873903</p></li>
<li><p>Bai X, Zheng Y, Hou L, Zheng C, Liu L, Zhang H. An Efficient
Testing Procedure for High-dimensional Mediators with FDR Control.
Statistics in Biosciences. 2024. DOI:
10.1007/s12561-024-09447-4.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Chicago</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] HIMA_2.3.1    glmnet_4.1-8  Matrix_1.7-1  ncvreg_3.14.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] gtable_0.3.6       shape_1.4.6.1      xfun_0.49          bslib_0.8.0       </span></span>
<span><span class="co">#&gt;  [5] ggplot2_3.5.1      htmlwidgets_1.6.4  lattice_0.22-6     vctrs_0.6.5       </span></span>
<span><span class="co">#&gt;  [9] tools_4.4.2        generics_0.1.3     parallel_4.4.2     tibble_3.2.1      </span></span>
<span><span class="co">#&gt; [13] fansi_1.0.6        pkgconfig_2.0.3    desc_1.4.3         linprog_0.9-4     </span></span>
<span><span class="co">#&gt; [17] lifecycle_1.0.4    compiler_4.4.2     stringr_1.5.1      MatrixModels_0.5-3</span></span>
<span><span class="co">#&gt; [21] textshaping_0.4.0  munsell_0.5.1      codetools_0.2-20   SparseM_1.84-2    </span></span>
<span><span class="co">#&gt; [25] quantreg_5.99.1    htmltools_0.5.8.1  sass_0.4.9         fdrtool_1.2.18    </span></span>
<span><span class="co">#&gt; [29] yaml_2.3.10        pillar_1.9.0       pkgdown_2.1.1      jquerylib_0.1.4   </span></span>
<span><span class="co">#&gt; [33] MASS_7.3-61        cachem_1.1.0       iterators_1.0.14   foreach_1.5.2     </span></span>
<span><span class="co">#&gt; [37] tidyselect_1.2.1   digest_0.6.37      stringi_1.8.4      dplyr_1.1.4       </span></span>
<span><span class="co">#&gt; [41] reshape2_1.4.4     splines_4.4.2      fastmap_1.2.0      grid_4.4.2        </span></span>
<span><span class="co">#&gt; [45] colorspace_2.1-1   cli_3.6.3          magrittr_2.0.3     survival_3.7-0    </span></span>
<span><span class="co">#&gt; [49] utf8_1.2.4         conquer_1.3.3      scales_1.3.0       rmarkdown_2.29    </span></span>
<span><span class="co">#&gt; [53] matrixStats_1.5.0  qvalue_2.36.0      hdi_0.1-9          ragg_1.3.3        </span></span>
<span><span class="co">#&gt; [57] HDMT_1.0.5         evaluate_1.0.1     lpSolve_5.6.23     knitr_1.49        </span></span>
<span><span class="co">#&gt; [61] scalreg_1.0.1      doParallel_1.0.17  lars_1.3           rlang_1.1.4       </span></span>
<span><span class="co">#&gt; [65] Rcpp_1.0.14        hommel_1.8         glue_1.8.0         rstudioapi_0.17.1 </span></span>
<span><span class="co">#&gt; [69] jsonlite_1.8.9     R6_2.5.1           plyr_1.8.9         systemfonts_1.1.0 </span></span>
<span><span class="co">#&gt; [73] fs_1.6.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Yinan Zheng, Haixiang Zhang, Lifang Hou, Lei Liu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
